utilities/logic/include_vars.py

```diff
@@ -10,28 +10,47 @@
 
 DOCUMENTATION = '''
 ---
-author: "Benno Joy (@bennojoy)"
+author: "Allen Sanabria (@linuxdynasty)"
 module: include_vars
 short_description: Load variables from files, dynamically within a task.
 description:
-     - Loads variables from a YAML/JSON file dynamically during task runtime.  It can work with conditionals, or use host specific variables to determine the path name to load from.
+     - Loads variables from a YAML/JSON files dynamically from within a file or
+       from a directory recursively during task runtime. If loading a directory, the files are sorted alphabetically before being loaded.
 options:
   file:
     version_added: "2.2"
     description:
        - The file name from which variables should be loaded.
        - If the path is relative, it will look for the file in vars/ subdirectory of a role or relative to playbook.
+  dir:
+    version_added: "2.2"
+    description:
+      - The directory name from which the variables should be loaded.
+      - If the path is relative, it will look for the file in vars/ subdirectory of a role or relative to playbook.
+    default: null
   name:
     version_added: "2.2"
     description:
         - The name of a variable into which assign the included vars, if omitted (null) they will be made top level vars.
     default: null
+  depth:
+    version_added: "2.2"
+    description:
+      - By default, this module will recursively go through each sub directory and load up the variables. By explicitly setting the depth, this module will only go as deep as the depth.
```

Comment:
Probably should somehow say that 0, the default, means it will recursively go through each subdirectory.  1 will only enter the directory itself, no subdirectories higher numbers go to an additional directorylevel. (Or you can change the code to default to None and use None to mean recurse all, 0 means only the directory itself).